src/components/cdPipeline/validationRules.ts

```diff
@@ -2,20 +2,29 @@ import { PATTERNS } from "../../config";
 
 export class ValidationRules {
 
-    name = (name: string, pattern?: string): { isValid: boolean, message: string } => {
-        if ((name.length)) {
-            let regex = new RegExp(pattern || PATTERNS.CD_PIPELINE_NAME);
-            if (regex.test(name)) return { isValid: true, message: null };
-            else return {
-                isValid: false,
-                message: `Must follow the pattern ${PATTERNS.CD_PIPELINE_NAME}`
-            };
+    name = (value: string, pattern?: string): { isValid: boolean, message: string } => {
+        let re="";
+        if(pattern){
+        re=pattern
         }
-        else return { isValid: false, message: 'This is a required field' };
+        else{
+         re = PATTERNS.APP_NAME
+        }
```

Comment:
We don't need this as we were already doing it with `new RegExp(pattern || PATTERNS.CD_PIPELINE_NAME)`. Please revert this and also confirm if we want all app name validations for cd pipeline name.